etcdserver/cluster_util.go

```diff
@@ -252,3 +252,12 @@ func getVersion(m *Member, tr *http.Transport) (*version.Versions, error) {
 	}
 	return nil, err
 }
+
+func MustDetectDowngrade(cv *semver.Version) {
+	lv := semver.Must(semver.NewVersion(version.Version))
+	// only keep major.minor version for comparison against cluster version
+	lv = &semver.Version{Major: lv.Major, Minor: lv.Minor}
+	if cv != nil && lv.LessThan(*cv) {
+		plog.Fatalf("cluster cannot be downgraded. (previous cluster version: %s, current version: %s)", version.Cluster(cv.String()), version.Cluster(lv.String()))
```

Comment:
I am not sure I understand what you mean exactly. Do you suggest replace the word `current version` to `server version`?  We never use `server version` and we always indicate the binary version as version. So I do not think so.